docs/content/basic/app-assistant.md

```diff
@@ -0,0 +1,156 @@
+    save: async ({ context, client, payload }) => {},
+  },
+  threadStarted: async ({ say, saveThreadContext, setStatus, setSuggestedPrompts, setTitle }) => {},
+  threadContextChanged: async ({ say, setStatus, setSuggestedPrompts, setTitle }) => {},
+  userMessage: async ({ say, getThreadContext, setStatus, setSuggestedPrompts, setTitle }) => {},
+});
+```
+
+You can store context through the `threadContextStore` property but it must feature `get` and `save` methods. If not provided, a `DefaultThreadContextStore` instance is utilized instead, which is a reference implementation that relies on storing and retrieving message metadata as the context changes.
+
+## Handling a new thread
+
+The `threadStarted` event handler allows your app to respond to new threads opened by users. In the example below, the app is sending a message — containing context message metadata — to the user, along with a single [prompt](https://api.slack.com/methods/assistant.threads.setSuggestedPrompts).
+
+```js
+...
+  threadStarted: async ({ event, say, setSuggestedPrompts, saveThreadContext }) => {
+    const { context } = event.assistant_thread;
+
+    await say({
+      text: 'Hi, how can I help?',
+      metadata: { event_type: 'assistant_thread_context', event_payload: context },
+    });
+
+    await saveThreadContext();
+
+    const prompts = [{
+      title: 'Fun Slack fact',
+      message: 'Give me a fun fact about Slack, please!',
+    }];
+
+    // Provide the user up to 4 optional, preset prompts to choose from.
+    await setSuggestedPrompts({ prompts });
+  },
+...
+```
+
+:::tip
+When a user opens an assistant thread while in a channel, the channel info is stored as the thread's `AssistantThreadContext` data. You can grab that info using the `getThreadContext()` utility, as subsequent user message event payloads won't include the channel info. 
+:::
+
+## Handling context changes 
+
+When the user switches channels, the `assistant_thread_context_changed` event will be sent to your app. Capture this with the `threadContextChanged` handler.
+
+```js
+...
+    threadContextChanged: async ({ saveThreadContext }) => {
+    await saveThreadContext();
+  },
+...
+```
+
+If you use the built-in `AssistantThreadContextStore` without any custom configuration the updated context data is automatically saved as message metadata on the first reply from the assistant bot.
+
+## Handling the user response
+
+User messages are handled with the `userMessage` event handler. The `setTitle` and `setStatus` utilities are useful in curating the user experience. 
+
+:::warning
+Messages sent to the assistant do not contain a subtype and must be deduced based on their shape and any provided metadata.
+:::
+
+ ```js
+ ...
+  userMessage: async ({ client, message, say, setTitle, setStatus }) => {
+    const { channel, thread_ts } = message;
+
+    // Set the title of the Assistant thread to capture the initial topic/question
+    await setTitle(message.text);
```

Comment:
Would this example change the title of the assistant view every time a user sends a message to the assistant view? @misscoded 🤔

Reply:
Yes it would. I didn't include set_title code in the python/java doc as it may not be relavant for many apps. Skipping this for introduction doc should be fine.